api/services/workflow_service.py

```diff
@@ -76,6 +78,38 @@ def get_published_workflow(self, app_model: App) -> Optional[Workflow]:
 
         return workflow
 
+    def get_all_published_workflow(self, app_model: App, page: int, limit: int) -> tuple[list[Workflow], bool]:
+        """
+        Get published workflow with pagination
+        """
+        if not app_model.workflow_id:
+            return [], False
+
+        workflows = (
+            db.session.query(Workflow)
+            .filter(Workflow.app_id == app_model.id)
+            .order_by(desc(Workflow.version))
+            .offset((page - 1) * limit)
+            .limit(limit + 1)
+            .all()
+        )
+
+        has_more = len(workflows) > limit
+        if has_more:
+            workflows = workflows[:-1]
+
+        if len(workflows) > 1:
+            workflows[1].version = "latest"
+
+        for workflow in workflows:
+            try:
+                version_datetime = datetime.strptime(workflow.version, "%Y-%m-%d %H:%M:%S.%f")
+                workflow.version = version_datetime.strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
update a db model for API response it's not a good way, please consider doing this convert in the front end.

The same issue is on L102.